### DISCUSSION TITLE
How to change number of decimal places printed for writeFabs()?

### DISCUSSION THREAD

**Discussion by ankithadas** [AMReX-Codes/amrex#3665] [Q&A] [answered]
Hey AMReX team!

I wanted to know how I could change the write precision of writeFabs() function. I use this function for debugging 

Any suggestions would be much appreciated! 

Kind regards,
Ankith

**Comment by cgilet** [comment]
In the inputs file, you can add "fab.format = " with any of these options with the "FAB_" removed (except for FAB_IEEE_32, which uses "IEEE32":
```
`    /**
    * \brief An enum which controls format of FAB output.
    *
    * FAB_ASCII: write the FAB out in ASCII format.
    *
    * FAB_8BIT: write the FAB out with all floating-point
    * values scaled to range 0 - 255.
    *
    * FAB_NATIVE: write out floating-point values in the native
    * format.  This is usually the "best" choice of formats.
    *
    * FAB_IEEE_32: write out floating-point values in IEEE 32
    * bit normal format.  This is recommended for use when your
    * internal computations are done in 64 bits and you want to save
    * space when writing out the FABs.
    *
    * FAB_IEEE: this is deprecated.  It is identical to
    * FAB_IEEE_32.
    *
    * FAB_NATIVE_32: write out values in the native 32 bit format.
    *
    */
    enum Format
    {
        FAB_ASCII,
        FAB_IEEE,
        FAB_NATIVE,
        //
        // This is set to four so that when reading in an old FAB,
        // we don't get confused when we see an old FAB_8BITRLE file.
        //
        FAB_8BIT = 4,
        FAB_IEEE_32,
        FAB_NATIVE_32
    };
` 
```

**Comment by WeiqunZhang** [comment, marked as answer]
I am curious. Why do you want to change the precision of the output, which is by default binary and double precision, for debugging?

**Reply by ankithadas** [reply]
Sorry I should have been a bit more specific. I wanted to change the precision of the FAB_ASCII format. I am using this for debugging and I would ideally want to read out the value.

**Reply by WeiqunZhang** [reply]
The precision of FAB_ASCII is determined by the stream's precision. You can copy the writeFabs function into your file and change the stream's precision by calling `precision` (https://en.cppreference.com/w/cpp/io/ios_base/precision). Note that if you want the full precision, set it to https://en.cppreference.com/w/cpp/types/numeric_limits/max_digits10.

**Reply by WeiqunZhang** [reply]
But I guess you want fewer digits.

**Reply by WeiqunZhang** [reply]
Note that you can also use `os << fab` to print fab in ASCII format. Probably more convenient than writeFabs. For the latter, you have to change the format to ASCII via inputs parameters or call a set function.

**Reply by ankithadas** [reply]
Thank you so much! That pretty much answers all my questions
